### PR TITLE
generalize comparators

### LINKED ISSUE #24: Generalize comparators
At the moment there is a bytes comparator, a number comparator and a random "comparator".

* [x] Create a new variable "comparator" of the same type as `bytes.Compare`
* [x] Make the other two comparators functions which fit into that variable
* [x] Initialize that variable once based on the given parameters and use it in the loop

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #24

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -66,46 +68,24 @@ func main() {
...
-				} else {
-					break
-				}
-			}
+		for j > 0 && comparator(dataSplit[j], dataSplit[j-1]) < 0 {
+			dataSplit[j], dataSplit[j-1] = dataSplit[j-1], dataSplit[j]
+			j -= 1
 		}
 	}
+
```
Comment (Al2Klimov) [inline main.go]: Not part of the task.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/generalize-comparators-24`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**Case(1):**
**toSort.txt:**
```
c
d
U
H
a
j
a
B
J
G
R
A
b
T
s
t
```

**GO CODE:**
**Terminal:** `go run . -R toSort.txt `
**Output:**
```
d
c
U
H
a
j
B
a
G
R
J
A
T
b
s
t
```

**SORT COMMAND:**
**Terminal:** `sort -R toSort.txt` 
**Output:**
```
R
a
a
U
c
s
J
B
T
H
A
d
b
j
t
G
```

**Case(2):**
**toSort.txt:**
```
10
3
5
4
6
4
3
67
74
3
26
3
541
1
```

**GO CODE:**
**Terminal:** `go run . -n toSort.txt `
**Output:**
```
1
3
3
3
3
4
4
5
6
10
26
67
74
541
```

**SORT COMMAND:**
**Terminal:** `sort -n toSort.txt` 
**Output:**
```
1
3
3
3
3
4
4
5
6
10
26
67
74
541
```

**Case(3):**
**toSort.txt:**
```
10
3
5
4
6
4
3
67
74
3
26
3
541
1
```

**GO CODE:**
**Terminal:** `go run . toSort.txt` 
**Output:**
```
1
10
26
3
3
3
3
4
4
5
541
6
67
74
```

**SORT COMMAND:**
**Terminal:** `sort toSort.txt `
**Output:**
```
1
10
26
3
3
3
3
4
4
5
541
6
67
74
```

**Conclusion:**
The program still behaves like before all of the changes.

**Pull request merged**